aten/src/ATen/core/ScalarType.h

```diff
@@ -38,6 +39,27 @@ enum class ScalarType {
   NumOptions
 };
 
+static inline DataType scalarTypeToDataType(ScalarType scalar_type) {
+#define DEFINE_CASE(ctype,name,_) \
+  case ScalarType:: name : return caffe2::TypeMeta::Id<ctype>();
+
+  switch(scalar_type) {
+    AT_FORALL_SCALAR_TYPES(DEFINE_CASE)
+    default: return DataType::uninitialized();
```

Comment:
what does uninitialized represent?  The asymmetry is a bit surprising here, I would guess both functions either throw or return some sentinel type.

Reply:
that sounds fine, just write that explicitly.